lib/utils/error-handler.js

```diff
@@ -17,6 +17,10 @@ var errorMessage = require('./error-message.js')
 process.on('exit', function (code) {
   log.disableProgress()
   if (!npm.config || !npm.config.loaded) return
+
+  // kill any outstanding stats reporter if it hasn't finished yet
+  if (npm.statsProcess) npm.statsProcess.kill('SIGKILL')
```

Comment:
`send-metrics` is explicitly written with the intent that killing it should be safe. I consider it more important that we be certain that there aren't any lingering processes (that are maybe updating the stats file) then that `send-metrics` not be interrupted.